lib/src/stereoCamera.cpp

```diff
@@ -567,80 +568,94 @@ void StereoCamera::computeDisparity(bool best, int uniquenessRatio, int speckleW
-        map.convertTo(map,CV_32FC1,255/(numberOfDisparities*16.));
-        //normalize(map,map, 0, 255, cv::NORM_MINMAX, CV_8UC1);
-
-    }
-    
-    if (cameraChanged)
+    	StereoSGBM sgbm;
+    	sgbm.preFilterCap =         preFilterCap; //63
+    	sgbm.SADWindowSize =        SADWindowSize;
+    	int cn =                    this->imleft.channels();
+    	sgbm.P1 =                   8*cn*sgbm.SADWindowSize*sgbm.SADWindowSize;
+    	sgbm.P2 =                   32*cn*sgbm.SADWindowSize*sgbm.SADWindowSize;
+    	sgbm.minDisparity =         minDisparity; //-15
+    	sgbm.numberOfDisparities =  numberOfDisparities;
+    	sgbm.uniquenessRatio =      uniquenessRatio; //22
+    	sgbm.speckleWindowSize =    speckleWindowSize; //100
+    	sgbm.speckleRange =         speckleRange; //32
+    	sgbm.disp12MaxDiff =        disp12MaxDiff;
+    	sgbm.fullDP =               best; // alg == STEREO_HH
+
+    	sgbm(img1r, img2r, disp);
+
+    	disp.convertTo(map, CV_32FC1, 1.0,0.0);
+    	map.convertTo(map,CV_32FC1,255/(numberOfDisparities*16.));
+    	//normalize(map,map, 0, 255, cv::NORM_MINMAX, CV_8UC1);
+
+    	success = true;
+    }
+
+    if (success)
     {
-        this->mutex->wait();
-        Mat inverseMapL(map.rows*map.cols,1,CV_32FC2);
-        Mat inverseMapR(map.rows*map.cols,1,CV_32FC2);
 
-        for (int y=0; y<map.rows; y++)
-        {
-            for (int x=0; x<map.cols; x++)
-            {
-                inverseMapL.ptr<float>(y*map.cols+x)[0]=(float)x;
-                inverseMapL.ptr<float>(y*map.cols+x)[1]=(float)y;
-                inverseMapR.ptr<float>(y*map.cols+x)[0]=(float)x;
-                inverseMapR.ptr<float>(y*map.cols+x)[1]=(float)y;
-            }
-        }
+    	if (cameraChanged)
+    	{
+    		this->mutex->wait();
 
-        undistortPoints(inverseMapL,inverseMapL,this->Kleft,this->DistL,this->RLrect,this->PLrect);
-        undistortPoints(inverseMapR,inverseMapR,this->Kright,this->DistR,this->RRrect,this->PRrect);
+    		Mat inverseMapL(map.rows*map.cols,1,CV_32FC2);
+    		Mat inverseMapR(map.rows*map.cols,1,CV_32FC2);
 
-        Mat mapperL=inverseMapL.reshape(2,map.rows);
-        Mat mapperR=inverseMapR.reshape(2,map.rows);
-        this->MapperL=mapperL;
-        this->MapperR=mapperR;
-        this->mutex->post();
-        cameraChanged=false;
-    }
+    		for (int y=0; y<map.rows; y++)
+    		{
+    			for (int x=0; x<map.cols; x++)
+    			{
+    				inverseMapL.ptr<float>(y*map.cols+x)[0]=(float)x;
+    				inverseMapL.ptr<float>(y*map.cols+x)[1]=(float)y;
+    				inverseMapR.ptr<float>(y*map.cols+x)[0]=(float)x;
+    				inverseMapR.ptr<float>(y*map.cols+x)[1]=(float)y;
+    			}
+    		}
 
-    Mat x;
-    remap(map,dispTemp,this->MapperL,x,cv::INTER_LINEAR);
-    dispTemp.convertTo(disp8,CV_8U); 
+    		undistortPoints(inverseMapL,inverseMapL,this->Kleft,this->DistL,this->RLrect,this->PLrect);
+    		undistortPoints(inverseMapR,inverseMapR,this->Kright,this->DistR,this->RRrect,this->PRrect);
```

Comment:
Oups .. sorry, there is indeed no change. The git diff made it look as if you swapped some commands, but in reality just some line breaks changed. Please ignore the previous comment :)